### ISSUE TITLE
0.12.0 - r.client.List does not compile due to dependency problem in go.sum

### ISSUE THREAD

**Issue by mhenke1** [operator-framework/operator-sdk#2164]
## Bug Report

**What did you do?**
Created an fresh operator with 
`operator-sdk new foo --repo github.ibm.com/bar/foo`

**What did you expect to see?**
* go.sum containing sigs.k8s.io/controller-runtime v0.3.0 only
* `r.client.List` should be callable with new signature expected with controller-runtime v0.3.0
* `operator sdk build `should succeed with a operator caling `r.client.List`

**What did you see instead? Under which circumstances?**
* go.sum does contain 
`sigs.k8s.io/controller-runtime v0.1.10/go.mod` as first entry for controller-runtime 
```
...
sigs.k8s.io/controller-runtime v0.1.10/go.mod h1:HFAYoOh6XMV+jKF1UjFwrknPbowfyHEHHRdJMf2jMX8=
sigs.k8s.io/controller-runtime v0.3.0 h1:ZtdgqJXVHsIytjdmDuk0QjagnzyLq9FjojXRqIp+dU4=
sigs.k8s.io/controller-runtime v0.3.0/go.mod h1:Cw6PkEg0Sa7dAYovGT4R0tRkGhHXpYijwNxYhAnAZZk=
...
```
* r.client.List still expects "old" signature
* `operator-sdk build` fails

**Environment**
* operator-sdk version:
0.12.0 (brew)

* go version:
go1.13.4 darwin/amd64

* Kubernetes version information:
Does not matter

* Kubernetes cluster kind: 
Does not matter

* Are you writing your operator in ansible, helm, or go?
go

**Possible Solution**
Add replacement clause 
`sigs.k8s.io/controller-runtime => sigs.k8s.io/controller-runtime v0.3.0` to go.mod

**Additional context**
go mod graph
```
github.com/operator-framework/operator-sdk@v0.12.0 github.com/operator-framework/operator-registry@v1.1.1
github.com/operator-framework/operator-lifecycle-manager@v0.0.0-20190605231540-b8a4faf68e36 github.com/operator-framework/operator-registry@v1.1.1
github.com/operator-framework/operator-registry@v1.1.1 sigs.k8s.io/controller-runtime@v0.1.10
```

**Comment by camilamacedo86** [comment]
HI @mhenke1, 

Thank you for raise it. You raised here 3 concerns/issues. Following the details and some questions. 

>NOTE: I try to check them in the master branch. After your answers, I will check the 0.12 as well.  


- [X] go.sum containing sigs.k8s.io/controller-runtime v0.3.0 only
This one I could reproduce/check it. OK

- [ ] r.client.List should be callable with new signature expected with controller-runtime v0.3.0

I created a new project + API + controller and could do it as follows. **Is it that you are trying to do? Could you provide your code impl?**

```go
	// Update the Memcached status with the pod names
	// List the pods for this memcached's deployment
	podList := &corev1.PodList{}
	listOpts := []client.ListOption{
		client.InNamespace(found.Namespace),
		client.MatchingLabels(labels),
	}
	err = r.client.List(context.TODO(), podList, listOpts...)
	if err != nil {
		reqLogger.Error(err, "Failed to list pods.", "Memcached.Namespace", found.Namespace, "Memcached.Name", found.Name)
		return reconcile.Result{}, err
	}

```


- [ ] operator sdk build should succeed with an operator calling r.client.List
I could build it successfully. **Could you provide the steps that are you following to reproduce some issue as the issue that you are facing?**

**Comment by joelanford** [comment]
@mhenke1 The `require` block in `go.mod` declares the minimum acceptable version of each of the listed dependencies for the module defined by that `go.mod`. So in this case, since controller-runtime `v0.3.0` is listed in your project's `go.mod` file, you are guaranteed that the resolved version will be at least `v0.3.0` and you are guaranteed that it will be less than `v1.0.0` since semver dictates that `v1` is not backwards compatible with `v0`.

This section of the Go wiki describes how this works: https://github.com/golang/go/wiki/Modules#version-selection

I also tried to reproduce the issue and was unable to.

**Comment by camilamacedo86** [comment]
I am closing this one as not reproducible. 
However, please feel free to re-open if you think that is required.
